File: Hard/2026-05-28-3093-Longest-Common-Suffix-Queries/solution.py
class Solution:
    def longestCommonSuffix(self, wordsContainer: List[str], wordsQuery: List[str]) -> List[int]:
        def common_suffix_length(s1: str, s2: str) -> int:
            l1, l2 = len(s1), len(s2)
            length = 0
            while length < l1 and length < l2 and s1[l1 - 1 - length] == s2[l2 - 1 - length]:
                length += 1
            return length
        
        ans = []
        for query in wordsQuery:
            max_length = -1
            best_index = -1
            for index, word in enumerate(wordsContainer):
                suffix_length = common_suffix_length(word, query)
                if (suffix_length > max_length or
                    (suffix_length == max_length and (best_index == -1 or len(wordsContainer[best_index]) > len(word)))):
                    max_length = suffix_length
                    best_index = index
            ans.append(best_index)
        
        return ans

File: Hard/2026-05-28-3093-Longest-Common-Suffix-Queries/solution.java
import java.util.*;

public class Solution {
    public int[] longestCommonSuffixQueries(String[] wordsContainer, String[] wordsQuery) {
        int n = wordsContainer.length;
        int[] result = new int[wordsQuery.length];
        
        for (int i = 0; i < wordsQuery.length; i++) {
            String query = wordsQuery[i];
            int maxLength = 0;
            int minLength = Integer.MAX_VALUE;
            int bestIndex = -1;

            for (int j = 0; j < n; j++) {
                String word = wordsContainer[j];
                int commonLength = commonSuffixLength(word, query);
                
                if (commonLength > maxLength || 
                    (commonLength == maxLength && word.length() < minLength) || 
                    (commonLength == maxLength && word.length() == minLength && j < bestIndex)) {
                    maxLength = commonLength;
                    minLength = word.length();
                    bestIndex = j;
                }
            }
            result[i] = bestIndex;
        }
        
        return result;
    }

    private int commonSuffixLength(String word, String query) {
        int wordLen = word.length();
        int queryLen = query.length();
        int length = 0;

        while (length < wordLen && length < queryLen && word.charAt(wordLen - 1 - length) == query.charAt(queryLen - 1 - length)) {
            length++;
        }
        
        return length;
    }
}